Dart/src/com/jetbrains/lang/dart/ide/runner/server/vmService/frame/DartVmServiceValue.java

```diff
@@ -149,10 +149,42 @@ public void computePresentation(@NotNull final XValueNode node, @NotNull final X
     if (computeRegExpPresentation(node)) return;
     if (computeMapPresentation(node)) return;
     if (computeListPresentation(node)) return;
-    computeDefaultPresentation(node);
+
+    // computeDefaultPresentation is called internally here when no result is got.
+    // The reason for this is that the async method used cannot be properly waited.
+    computeDebugPresentation(node);
+
     // todo handle other special kinds: Type, TypeParameter, Pattern, may be some others as well
   }
 
+  private void computeDebugPresentation(final XValueNode node) {
+    myDebugProcess.getVmServiceWrapper()
+      .evaluateInTargetContext(myIsolateId, myInstanceRef.getId(), "toStringDeep()", new VmServiceConsumers.EvaluateConsumerWrapper() {
+        @Override
+        public void received(final InstanceRef toStringInstanceRef) {
+          if (toStringInstanceRef.getKind() == InstanceKind.String) {
+            String content = toStringInstanceRef.getValueAsString();
+            String summary = content.split("\n")[0];
```

Comment:
I usually try to avoid regexps in simple cases for performance reasons. Here looks like it can be replaced with `substring()` + `indexOf()`.

Reply:
Good suggestion. Thanks!